test/unit/DCAStrategies/dca-strategies-permissions-handler.spec.ts

```diff
@@ -77,22 +77,9 @@ contract('DCAStrategiesPermissionsHandler', () => {
         expect(await DCAStrategiesPermissionsHandlerMock.mintCounter()).to.equal(initialMintCounter.add(1));
       });
 
-      //   then('owner has all permisisons', async () => {
-      //     expect(await DCAStrategiesPermissionsHandlerMock.hasPermission(TOKEN_ID, OWNER, Permission.INCREASE)).to.be.true;
-      //     expect(await DCAStrategiesPermissionsHandlerMock.hasPermission(TOKEN_ID, OWNER, Permission.REDUCE)).to.be.true;
-      //     expect(await DCAStrategiesPermissionsHandlerMock.hasPermission(TOKEN_ID, OWNER, Permission.TERMINATE)).to.be.true;
-      //     expect(await DCAStrategiesPermissionsHandlerMock.hasPermission(TOKEN_ID, OWNER, Permission.WITHDRAW)).to.be.true;
-      //   });
-
-      //   then('permissions are assigned properly', async () => {
-      //     expect(await DCAStrategiesPermissionsHandlerMock.hasPermission(TOKEN_ID, OPERATOR, Permission.WITHDRAW)).to.be.true;
-      //   });
-
-      //   then('no extra permissions are assigned', async () => {
-      //     for (const permission of [Permission.INCREASE, Permission.REDUCE, Permission.TERMINATE]) {
-      //       expect(await DCAStrategiesPermissionsHandlerMock.hasPermission(TOKEN_ID, OPERATOR, permission)).to.be.false;
-      //     }
-      //   });
+      then('permissions are assigned properly', async () => {
```

Comment:
Since you prefer to go with option 2, I recommend you do something like [this](https://github.com/Mean-Finance/dca-v2-core/blob/main/contracts/mocks/DCAPermissionsManager/DCAPermissionsManager.sol#L22)
1. In the mock you override the actuall call to `_setPermissions`
2. You register all calls that went through it
3. You then expose a way to get all calls
4. In `mint` you make sure that it was called once, and with the correct parameters